README.md

```diff
@@ -3,16 +3,43 @@
 osp_templates
 ===========
 
-Basic description for osp_templates
+Ansible role to template an entire directory structure efficiently. It can be operated
+in different modes to template either the entire structure or to leverage git to only
+template files which have changed. Rsync is used to drop the final files into place
+once the templating step is complete
+
+The role can operate in one of three modes. It has a "template all" mode, where
+it walks the `osp_templates_input_dir` and processes each file through the Jinja2 filter
+system and duplicates it into the `osp_templates_output_dir` structure. It can also
+operate in a "quick" mode where it will do the same, but it will only operate on files
+that are reported as changed by a git status. This will require that "git" be installed
+on the Ansible operator node. The third mode is "copy only" mode, where rsync is used
+to synchronize files from input to output directories. This requires rsync be installed
+and that any hosts be accessible (without password arguments). For requirements on the
+synchronize, check the [synchronize module](https://docs.ansible.com/ansible/latest/modules/synchronize_module.html#synchronize-module)
+in Ansible.
+
+The default mode is "template all". To enable quick mode set `osp_templates_quick_mode`
+to true. This will avoid the extra time of evaluating every template. To enable copy
+mode, set `osp_templates_copy_only` to true. Only one mode at a time can be executed.
+See below for information on different ways to use the role multiple times if you need
+to run in multiple steps.
+
+There is a very extensive set of sample files in the `samples/` directory that should
+be useful in deploying a [Red Hat OpenStack Platform](https://www.redhat.com/en/technologies/linux-platforms/openstack-platform)
+instance with Director. These files are not necessarily going to be perfectly suited to
+every environment or system. Feel free to grab those files, update them as appropriate,
+and specify your own directory as the `osp_templates_input_dir` variable. Also, there is
+nothing in this role, besides the samples, that is specifically tied to OpenStack.
```

Comment:
If the role itself isn't specifically tied to openstack, is there any interest in changing the "osp_templates" name to also not be tied to openstack?

Reply:
I've thought of that. It's definitely an OSP role, though. At least with its samples directory the way it is.